rspec-support/lib/rspec/support/ruby_features.rb

```diff
@@ -96,6 +96,16 @@ def supports_exception_cause?
         end
       end
 
+      if Exception.method_defined?(:detailed_message)
+        def supports_exception_detailed_message?
+          true
+        end
+      else
+        def supports_exception_detailed_message?
+          false
+        end
+      end
```

Comment:
Can you break this out into a separate PR, the only reasoning I have for this is that it will allow me to merge it so we don't have to keep approving your workflows.

Reply:
Done on https://github.com/rspec/rspec/pull/203